targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,353 @@
+    {
+        private class CustomSerializerPlugin : ISerializerPlugin
+        {
+            public bool IsSerializeCalled = false;
+
+            public T DeserializeObject<T>(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public T DeserializeObject<T>(string serialized, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object DeserializeObject(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public string SerializeObject(object obj)
+            {
+                this.IsSerializeCalled = true;
+                return "{}";
+            }
+
+            public string SerializeObject(object obj, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class CustomTransportPlugin : ITransportPlugin
+        {
+            public string Name;
+            public bool IsDoPostCalled = false;
+
+            public async Task<object> DoPost(string urlPath, object request, Dictionary<string, string> headers)
+            {
+                this.IsDoPostCalled = true;
+                return "{}";
+            }
+        }
+
+        private class CustomApi
+        {
+            public string PluginName;
+
+            public async Task<object> SomeApiMethod()
+            {
+                var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport, this.PluginName);
+                return await transport.DoPost(null, null, null);
+            }
+        };
+
+        // Functional
+        private static bool TITLE_INFO_SET = false;
+
+        // Fixed values provided from testInputs
+        private static string USER_EMAIL;
+        private static Dictionary<string, string> extraHeaders;
+
+        /// <summary>
+        /// PlayFab Title cannot be created from SDK tests, so you must provide your titleId to run unit tests.
+        /// (Also, we don't want lots of excess unused titles)
+        /// </summary>
+        public static void SetTitleInfo(TestTitleData testInputs)
+        {
+            TITLE_INFO_SET = true;
+
+            PlayFabSettings.TitleId = testInputs.titleId;
+            USER_EMAIL = testInputs.userEmail;
+            extraHeaders = testInputs.extraHeaders;
+
+            // Verify all the inputs won't cause crashes in the tests
+            TITLE_INFO_SET &= !string.IsNullOrEmpty(PlayFabSettings.TitleId)
+                && !string.IsNullOrEmpty(USER_EMAIL);
+        }
+
+        public override void SetUp(UUnitTestContext testContext)
+        {
+            if (!TITLE_INFO_SET)
+                testContext.Skip(); // We cannot do client tests if the titleId is not given
+        }
+
+        public override void Tick(UUnitTestContext testContext)
```

Comment:
Empty override functions aren't actually necessary.
You can keep them if you think you will use them, but if you know you won't need them, you should probably just delete it.